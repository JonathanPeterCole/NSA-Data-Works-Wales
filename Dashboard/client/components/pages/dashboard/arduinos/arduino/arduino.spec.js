import React from 'react'
import { shallow } from 'enzyme'

// Component
import Arduino from './arduino'

// Prepare a test Arduino
const testArduino = {
  name: 'Test Arduino',
  colour: 'blue',
  online: true,
  lastConnection: 0,
  sensors: [
    {
      id: 0,
      type: 'temperature',
      online: true,
      readings: []
    }
  ]
}

// Prepare a mock onClick function
const mockOnClick = jest.fn()

// Tests
describe('Arduino Component', () => {
  beforeEach(() => {
    // Clear mockOnClick before each test
    mockOnClick.mockClear()
  })

  it('Renders without crashing', () => {
    // Attempt to render the Arduino component
    // Shallow render to ensure we are only testing the Arduino component
    shallow(
      <Arduino
        name={testArduino.name}
        colour={testArduino.colour}
        online={testArduino.online}
        lastConnection={testArduino.lastConnection}
        sensors={testArduino.sensors}
        onClick={mockOnClick} />
    )
  })

  it('Matches snapshot', () => {
    // Compare the component to the snapshot
    const wrapper = shallow(
      <Arduino
        name={testArduino.name}
        colour={testArduino.colour}
        online={testArduino.online}
        lastConnection={testArduino.lastConnection}
        sensors={testArduino.sensors}
        onClick={mockOnClick} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Calls the open modal function on click', () => {
    // Render the component
    const wrapper = shallow(
      <Arduino
        name={testArduino.name}
        colour={testArduino.colour}
        online={testArduino.online}
        lastConnection={testArduino.lastConnection}
        sensors={testArduino.sensors}
        onClick={mockOnClick} />
    )
    // Simulate a click on the Arduino box
    wrapper.find('div.box').first().simulate('click')
    // Check if the mockOnClick was called
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
})
