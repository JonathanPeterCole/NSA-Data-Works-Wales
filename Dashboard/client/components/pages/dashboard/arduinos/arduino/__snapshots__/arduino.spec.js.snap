// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arduino Component Matches snapshot 1`] = `
<div
  className="arduino"
>
  <div
    className="box blue"
    onClick={[MockFunction]}
  >
    <div
      className="arduino-header"
    >
      <div
        className="title"
      >
        Test Arduino
      </div>
      <StatusIndicator
        lastConnection={0}
        online={true}
      />
    </div>
    <div
      className="arduino-readings"
    >
      <Reading
        key="0"
        name="Temperature 0"
        online={true}
        readings={Array []}
        type="temperature"
      />
    </div>
  </div>
</div>
`;
